src/Utils/client.js

```diff
@@ -0,0 +1,71 @@
+import axios from "axios";
+const API_BASE = "https://pokeapi.co/api/v2";
+const POKEMON_TOTAL = 1292; // setting limit for "all pokemon" query
+
+/////////////
+// HELPERS //
+/////////////
+
+/////////////
+// PRUNING // (since we don't want thaaaat much data)
+/////////////
+
+const pruneTypes = (typesData) => {
+  return typesData.map((t) => t.type.name);
+};
+
+const pruneSprites = (spritesData) => {
+  const prunedData = {};
+
+  // this field seemed suspect so this is a cautious measure
+  if (
+    spritesData["other"] !== undefined &&
+    spritesData["other"]["official-artwork"] !== undefined
+  ) {
+    prunedData.large =
+      spritesData["other"]["official-artwork"]["front_default"];
+  }
+  prunedData.small = spritesData["front_default"];
+
+  return prunedData;
+};
+
+const prunePokemon = (pokemonData) => {
+  const prunedData = {};
+  prunedData.id = pokemonData.id;
+  prunedData.name = pokemonData.name;
+  prunedData.height = parseInt(pokemonData.height) / 10; // decimeters -> meters
+  prunedData.weight = parseInt(pokemonData.weight) / 10; // hectograms -> kilograms
+  prunedData.types = pruneTypes(pokemonData.types);
+  prunedData.sprites = pruneSprites(pokemonData.sprites);
+
+  return prunedData;
+};
+
+/////////////
+// EXPORTS //
+/////////////
+
+export const findPokemon = async (name) => {
+  const response = await axios.get(`${API_BASE}/pokemon/${name}`);
+  if (response.data === undefined) {
+    return undefined;
+  }
+
+  return prunePokemon(response.data);
+};
+
+// GET: list of Pokemon names (not full Pokemon data)
+// not case-sensitive
+export const searchPokemonBySubstring = async (substring) => {
+  const response = await axios.get(
+    `${API_BASE}/pokemon?limit=${POKEMON_TOTAL}`
+  ); // all pokemon
+
+  const allPokemon = response.data.results;
+  const filteredPokemon = allPokemon
+    .filter((pokemon) => pokemon.name.includes(substring.toLowerCase())) // has search query in name
```

Comment:
is this enough for "how should searching actually work", or do we want subsequence, which i can implement fairly quickly

Reply:
most pokemon are 1 word names so i think this is good enough